### ISSUE TITLE
RF-DETR Segmentation to ONNX

### ISSUE THREAD

**Issue by AICTIControl** [roboflow/rf-detr#463]
### Search before asking

- [x] #464

### Bug

When I train using `rfdetr` library, I got excelent results on my predictions and the masks seems to be smooth, as it should be. 

However, when I export my RF-DETR Segmentation model (Preview) to ONNX for edge inference, I got good predictions, but poor masks detection.

As you can see in the conversion below, Mask output dimensions are [1, 2600, 150, 150], which means is an image with size 150 * 150. I'm convinced this is the problem, and due to this, the masks prediction from the outputs of ONNX model fit the object, although it has aliasing edges.

> <img width="267" height="203" alt="Image" src="https://github.com/user-attachments/assets/ddf20fb4-ee80-4493-a108-bc0db0c1969f" />
> 
> <img width="267" height="193" alt="Image" src="https://github.com/user-attachments/assets/35077d0e-6a76-4287-a6d0-4700a5549e7c" />

The previous images shows the contours of a mask for an object detected using ONNX model (left) and base rfdetr library (working on PyTorch, on the right). Keeping this in mind, Is it possible to fix the ONNX output in order to smooth the masks for ONNX exporting model? Is this issue related with mask image size of outputs? 

### Environment

1. RF-DETR: 1.3.0
2. OS: Windows 11
3. Python: 3.10.16
4. PyTorch: 2.8.0+cu129
5. CUDA: 12.9
6. GPU: NVIDIA RTX 4080 Super

### Minimal Reproducible Example

```python
model = rfdetr.RFDETRSegPreview(pretrain_weights='outputs/checkpoint_best_ema.pth', resolution=384)
model.export(output_dir="onnx", infer_dir=None, simplify=False,  backbone_only=False)
```
```bash
Exporting model to ONNX format
PyTorch inference output shapes - Boxes: torch.Size([1, 2600, 4]), Labels: torch.Size([1, 2600, 3]), Masks: torch.Size([1, 2600, 150, 150])
TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
TracerWarning: torch.as_tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
TracerWarning: Iterating over a tensor might cause the trace to be incorrect. Passing a tensor of different shape won't change the number of iterations executed (and might lead to errors or silently give incorrect results).
TracerWarning: torch.tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
TracerWarning: torch.tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
TracerWarning: Iterating over a tensor might cause the trace to be incorrect. Passing a tensor of different shape won't change the number of iterations executed (and might lead to errors or silently give incorrect results).
TracerWarning: Iterating over a tensor might cause the trace to be incorrect. Passing a tensor of different shape won't change the number of iterations executed (and might lead to errors or silently give incorrect results).

Successfully exported ONNX model: onnx\inference_model.onnx
Successfully exported ONNX model to: onnx\inference_model.onnx
ONNX export completed successfully
```

### Are you willing to submit a PR?

- [ ] Yes, I'd like to help by submitting a PR!

**Comment by isaacrob-roboflow** [comment]
there may be subtle differences between different pipelines but the mAP should be approximately the same. if it's not there's a bug. but we export to ONNX using the existing infra and tend to see no significant change

**Comment by 1chengz** [comment]
I used the same random input for both the PT model and its corresponding ONNX model, and the output results can only be described as similar but not identical, primarily reflected in the mask section

![Image](https://github.com/user-attachments/assets/1bab0952-5dee-4fae-aeb2-f68f9e1d98af)

**Comment by Borda** [comment]
Hi @AICTIControl — apologies for the delayed follow-up. The 150×150 mask output in ONNX is expected; masks are exported at a compressed resolution for efficiency. As @isaacrob advised, bilinearly upsample the mask logits to your target resolution before thresholding. Minor numerical differences between PT and ONNX are normal due to backend differences — mAP should be approximately equivalent. v1.6.1 deprecated the simplify argument (#861), and #871 (dynamic batch ONNX export, merged to develop, coming in the next release) brings further improvements. Closing this as the behavior is by-design with a known workaround.
